netplugin/plugin/netplugin.go

```diff
@@ -303,27 +304,57 @@ func (p *NetPlugin) Reinit(cfg Config) {
 }
 
 //InitGlobalSettings initializes cluster-wide settings (e.g. fwd-mode)
-func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) {
+func InitGlobalSettings(stateDriver core.StateDriver, inst *core.InstanceInfo) error {
+
+	/*
+	Query global settings from state store
+	1. if forward mode or private net is empty, retry 1 sec
+	2. if forward mode doesn't match, return error
+	3. if stored private net is wrong, return error
+	*/
 
 	gCfg := mastercfg.GlobConfig{}
 	gCfg.StateDriver = stateDriver
-	err := gCfg.Read("")
-	if err != nil {
-		logrus.Errorf("Error reading forwarding mode from cluster store")
-		inst.FwdMode = "bridge"
-		inst.HostPvtNW = defaultPvtSubnet
-		return
+
+	// wait until able to get fwd mode and private subnet
+	for {
+		err := gCfg.Read("")
+		if err != nil {
+			logrus.Errorf("Error reading global settings from cluster store")
+		}
+		if gCfg.FwdMode == "" || gCfg.PvtSubnet == "" {
+			if gCfg.FwdMode == "" {
+				logrus.Errorf("Error reading forwarding mode from cluster store")
+			} else {
+				logrus.Errorf("Error reading private subnet cluster store")
+			}
+
+		} else {
+			logrus.Infof("Get global forwarding mode: %v", gCfg.FwdMode)
+			logrus.Infof("Get global private subnet: %v", gCfg.PvtSubnet)
+			break
+		}
+		logrus.Infof("Sleep 1 second and retry pulling global settings")
+		time.Sleep(1 * time.Second)
+	}
+
+	if inst.FwdMode == "" {
+		inst.FwdMode = gCfg.FwdMode
+	} else {
+		if inst.FwdMode != gCfg.FwdMode {
+			logrus.Errorf("Global forwarding mode setting doesn't match local settings, global=%v , local=%v", gCfg.FwdMode, inst.FwdMode)
+			return core.Errorf("Global forwarding mode setting doesn't match local settings, global=%v , local=%v", gCfg.FwdMode, inst.FwdMode)
```

Comment:
Can we use fmt.Errorf and assign it to a variable instead of having two duplicated string?

Reply:
it's actually every odd we had core.Errorf and its logic .... yeah I will change to use fmt.Errorf , but  codes with core.Errorf has been used in many places through the project

Reply:
@dseevr  ^ we need to pick one way or the other and keep it consistent

Reply:
or use `errors` pkg

Reply:
We should get rid of `core.Error` entirely IMO.  We don't need the stack trace stuff in it.

Based on the people who wrote it (erik/madhav) and the dates, it looks like it was a precursor to [errored](https://github.com/contiv/errored).

I'll add an issue for this tomorrow

Reply:
I'd say let's use `fmt.Errorf()` or `errors.New()` for now.  As Kahou said, let's also avoid duplicating strings.  There's a lot of those that need to be cleaned up in the codebase(s)...

Reply:
done